### DISCUSSION TITLE
Is there a way to generate code from schema instead (schema-first)?

### DISCUSSION THREAD

**Discussion by frederikhors** [async-graphql/async-graphql#913] [Q&A]
I'm new to Rust.

I come from Go and [gqlgen](https://gqlgen.com/).

Is there a way to generate code from schema like in [gqlgen](https://gqlgen.com/)?

Your job is amazing! Thank you very much for this!

**Comment by simoneromano96** [comment]
Not currently, I remember juniper had a tool to work schema-first however.

https://github.com/davidpdrsn/juniper-from-schema

Maybe we could analyze this project and port it to async-graphql?

**Comment by sunli829** [comment]
[codegen-for-async-graphql](https://crates.io/crates/codegen-for-async-graphql)

but unfortunately only a small part of it has been completed.

**Reply by frederikhors** [reply]
Thanks. Aren't you interested in this? Don't you see value in code-first?

Imagine a company that has to migrate a huge scheme, this could help them choose async-graphql.

**Reply by sunli829** [reply]
I totally understand, I just think `Code-first` is more convenient. 🙂

**Reply by kennetpostigo** [reply]
@frederikhors Rust type system is very expressive and translates very well to GraphQL. You could attempt a small part of your schema with the Code first approach and see how it goes.
